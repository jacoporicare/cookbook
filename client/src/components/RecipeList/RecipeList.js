import React from 'react';
import PropTypes from 'prop-types';
import RecipeListItem from './RecipeListItem';

const RecipeList = ({ recipes }) => (
  <div className="row">
    {recipes.map(recipe => <RecipeListItem recipe={recipe} key={recipe._id} />)}
    {!recipes.length && (
      <div className="alert alert-info">Zatím zde není žádný recept.</div>
    )}
  </div>
);

RecipeList.propTypes = {
  recipes: PropTypes.array.isRequired,
};

export default RecipeList;
