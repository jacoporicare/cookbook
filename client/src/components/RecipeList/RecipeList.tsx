import React from 'react';

import { Recipe } from '../../types';
import { RecipeListItem } from './RecipeListItem';

type Props = {
  recipes: Recipe[];
};

export const RecipeList = ({ recipes }: Props) => (
  <div className="row">
    {recipes.map(recipe => (
      <RecipeListItem key={recipe._id} recipe={recipe} />
    ))}
    {recipes.length === 0 && <div className="alert alert-info">Zatím zde není žádný recept.</div>}
  </div>
);
