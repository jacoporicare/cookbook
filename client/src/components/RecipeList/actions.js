import { CALL_API } from '../../middleware/api';

export const RECIPE_LIST_FETCH_REQUEST = 'RECIPE_LIST.FETCH.REQUEST';
export const RECIPE_LIST_FETCH_SUCCESS = 'RECIPE_LIST.FETCH.SUCCESS';
export const RECIPE_LIST_FETCH_FAILURE = 'RECIPE_LIST.FETCH.FAILURE';

export const fetchRecipeListRequest = () => ({
  type: RECIPE_LIST_FETCH_REQUEST,
});

export const fetchRecipeListSuccess = recipes => ({
  type: RECIPE_LIST_FETCH_SUCCESS,
  payload: {
    recipes,
  },
});

export const fetchRecipeListFailure = (errorMessage, response) => ({
  type: RECIPE_LIST_FETCH_FAILURE,
  payload: {
    errorMessage,
    response,
  },
});

export const fetchRecipeList = () => ({
  [CALL_API]: {
    actions: [
      fetchRecipeListRequest,
      fetchRecipeListSuccess,
      fetchRecipeListFailure,
    ],
    url: '/api/recipes',
  },
});
