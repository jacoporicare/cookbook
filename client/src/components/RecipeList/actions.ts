import { Dispatch } from 'redux';

import { Recipe, StoreState } from '../../types';
import api, { handleError } from '../../api';

export type RecipeListAction =
  | { type: 'RECIPE_LIST.FETCH.REQUEST' }
  | { type: 'RECIPE_LIST.FETCH.SUCCESS'; payload: { recipes: Recipe[] } }
  | { type: 'RECIPE_LIST.FETCH.FAILURE' };

export const fetchRecipeListRequest = (): RecipeListAction => ({
  type: 'RECIPE_LIST.FETCH.REQUEST',
});

export const fetchRecipeListSuccess = (recipes: Recipe[]): RecipeListAction => ({
  type: 'RECIPE_LIST.FETCH.SUCCESS',
  payload: {
    recipes,
  },
});

export const fetchRecipeListFailure = (): RecipeListAction => ({
  type: 'RECIPE_LIST.FETCH.FAILURE',
});

export function fetchRecipeList() {
  return (dispatch: Dispatch<StoreState>, getState: () => StoreState) => {
    dispatch(fetchRecipeListRequest());

    return api(getState)
      .get<Recipe[]>('/api/recipes/')
      .then(({ data }) => dispatch(fetchRecipeListSuccess(data)))
      .catch(error => {
        handleError(error);
        return dispatch(fetchRecipeListFailure());
      });
  };
}
