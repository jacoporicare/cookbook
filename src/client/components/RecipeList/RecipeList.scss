@import '~bootstrap-sass/assets/stylesheets/bootstrap/variables';

$cb-recipe-spacing: 15px;
$cb-recipe-border-color: #f0f0f0;

.cb-recipes {
  .recipe {
    display: block;
    border: 1px solid $cb-recipe-border-color;
    border-radius: 2px;
    box-shadow: 0 0 12px 2px $cb-recipe-border-color;
    margin-bottom: ($cb-recipe-spacing * 2);
    padding: $cb-recipe-spacing;
    text-decoration: none;

    &:hover {
      box-shadow: 0 0 12px 2px darken($cb-recipe-border-color, 10%);
    }

    h4 {
      margin: 0;
      font-weight: bold;
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;
    }

    .cb-info-list {
      margin-top: 1em;
    }
  }
}
