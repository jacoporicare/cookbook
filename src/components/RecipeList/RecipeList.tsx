import React from 'react';

import { Recipe } from '../../types';
import { colors } from '../../styles/colors';
import { BoxArticle, Box } from '../core';

import RecipeListItem from './RecipeListItem';

type Props = {
  recipes: Recipe[];
};

const widths = ['100%', '50%', '33.33333%', '25%'];
const border = `1px solid ${colors.gray200}`;

function RecipeList({ recipes }: Props) {
  return (
    <Box
      borderLeft={[0, border]}
      borderTop={[0, border]}
      display="flex"
      flexWrap="wrap"
      justifyContent="flex-start"
    >
      {recipes
        .slice()
        .sort((a, b) => a.title.localeCompare(b.title, 'cs'))
        .map(recipe => (
          <BoxArticle
            key={recipe._id}
            borderBottom={border}
            borderLeft={[border, 0]}
            borderRight={border}
            borderTop={[border, 0]}
            flex="auto"
            maxWidth={widths}
            mb={[3, 0]}
            width={widths}
          >
            <RecipeListItem recipe={recipe} />
          </BoxArticle>
        ))}
    </Box>
  );
}

export default RecipeList;
