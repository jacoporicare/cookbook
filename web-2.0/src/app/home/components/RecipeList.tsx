import { Grid } from '@mui/material';

import RecipeListItem from './RecipeListItem';

import { RecipeListQuery } from '@/gql/graphql';

type Props = {
  recipes: RecipeListQuery['recipes'];
};

export default async function RecipeList(props: Props) {
  return (
    <main>
      <Grid spacing={2} container>
        {props.recipes.map(recipe => (
          <Grid key={recipe.id} component="article" lg={3} md={4} sm={6} xl={2} xs={12} item>
            <RecipeListItem recipe={recipe} />
          </Grid>
        ))}
      </Grid>
    </main>
  );
}
