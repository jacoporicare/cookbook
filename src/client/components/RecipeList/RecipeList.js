import React, { PropTypes } from 'react';
import RecipeListItem from './RecipeListItem';
import './RecipeList.scss';

const RecipeList = ({ recipes }) => {
  return (
    <div className="cb-recipes">
      <div className="row">
        {recipes.map(recipe =>
          <RecipeListItem recipe={recipe} key={recipe._id} />
        )}
      </div>
      {!recipes.length &&
        <div className="alert alert-info">
          Zatím zde není žádný recept.
        </div>
      }
    </div>
  );
};

RecipeList.propTypes = {
  recipes: PropTypes.array.isRequired
};

export default RecipeList;
