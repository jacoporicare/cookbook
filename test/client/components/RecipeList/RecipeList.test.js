import React from 'react';
import renderer from 'react-test-renderer';
import RecipeList from '../../../../src/client/components/RecipeList/RecipeList';

describe('<RecipeList />', () => {
  test('renders recipes', () => {
    const recipes = [
      {
        _id: 1,
        slug: 'recept-1',
        title: 'Recept 1',
      },
      {
        _id: 2,
        slug: 'recept-2',
        title: 'Recept 2',
        preparationTime: 120,
      },
      {
        _id: 3,
        slug: 'recept-3',
        title: 'Recept 3',
        sideDish: 'kaše',
      },
      {
        _id: 4,
        slug: 'recept-4',
        title: 'Recept 4',
        preparationTime: 120,
        sideDish: 'kaše',
      },
    ];

    const component = renderer.create(
      <RecipeList recipes={recipes} />
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
