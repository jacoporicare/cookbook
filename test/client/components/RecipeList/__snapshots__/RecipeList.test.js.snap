exports[`<RecipeList /> renders recipes 1`] = `
<div
  className="row cb-recipes">
  <div
    className="col-sm-6 col-md-4">
    <a
      className="recipe"
      onClick={[Function]}
      style={Object {}}>
      <h4
        className="text-primary">
        Recept 1
      </h4>
    </a>
  </div>
  <div
    className="col-sm-6 col-md-4">
    <a
      className="recipe"
      onClick={[Function]}
      style={Object {}}>
      <h4
        className="text-primary">
        Recept 2
      </h4>
      <ul
        className="cb-info-list text-muted">
        <li>
          <i
            className="fa fa-clock-o" />
           
          120
           min
        </li>
      </ul>
    </a>
  </div>
  <div
    className="col-sm-6 col-md-4">
    <a
      className="recipe"
      onClick={[Function]}
      style={Object {}}>
      <h4
        className="text-primary">
        Recept 3
      </h4>
      <ul
        className="cb-info-list text-muted">
        <li>
          <i
            className="fa fa-spoon" />
           
          kaše
        </li>
      </ul>
    </a>
  </div>
  <div
    className="col-sm-6 col-md-4">
    <a
      className="recipe"
      onClick={[Function]}
      style={Object {}}>
      <h4
        className="text-primary">
        Recept 4
      </h4>
      <ul
        className="cb-info-list text-muted">
        <li>
          <i
            className="fa fa-clock-o" />
           
          120
           min
        </li>
        <li>
          <i
            className="fa fa-spoon" />
           
          kaše
        </li>
      </ul>
    </a>
  </div>
</div>
`;
